### ISSUE TITLE
Travis CI build stalling after `before_script` for todo_challenge

### ISSUE THREAD

**Issue by trwh** [makersacademy/slack-overflow#124]
Following on from my [previous question](https://github.com/makersacademy/slack-overflow/issues/123), my CI tests now stall before running my test scripts.

Here is the failing build:

https://travis-ci.org/makersacademy/todo_challenge/builds/87049442

```
15:00:25.117 INFO - Selenium Server is up and running
[END OF OUTPUT, REPORT FAILED BUILD AFTER 10 MINUTES DELAY]
```

Here is my `.travis.yml`,

```yml
script:
  # - npm test
  # - node_modules/.bin/protractor test/e2e/conf.js
  - npm run test-karma
  - npm run test-protractor
language: node_js
node_js:
  - "0.10"
before_script:
  - bower install
  - "export DISPLAY=:99.0"
  - "sh -e /etc/init.d/xvfb start"
  - webdriver-manager update
  - webdriver-manager start > /dev/null &
  - npm start > /dev/null &
  - http-server
```

I've tried both `npm test` and `npm run test-karma` `npm run test-protractor` with no luck. To confirm, both of these tests work locally and are `--single-run` so don't hang the console for ever when run. For interest, here is the relevant part of my `package.json` file:

```json
"scripts": {
    "test": "npm run test-karma && npm run test-protractor",
    "test-karma": "karma start test/karma.conf.js --single-run",
    "test-protractor": "protractor test/e2e/conf.js"
  }
```

As ever, I really appreciate any help!

Tim

**Comment by mechacoach** [comment]
A word to the wise - format overflow questions like this:

1. What you're trying to do
2. The code to do it in a [GitHub-flavoured Markdown](https://help.github.com/articles/github-flavored-markdown/) code block
3. The error you're getting

That will help a casual browser to quickly point you in the right direction.

**Comment by spike01** [comment]
`http-server` is blocking the process - look at how it's been dealt with after `npm start` (I know I'm being deliberately vague! Let me know if that's totally useless)

**Comment by trwh** [comment]
Many thanks! I updated the launch of `http-server` to include `> /dev/null &`, that is, to pipe the output to null and background the task and it works. :)

However, I couldn't get Chrome working on Travis despite having included `"karma-chrome-launcher": "^0.2.1"`. I don't think you guys could either as you use Firefox in the supplied `.travis.yml`. So I included `"karma-firefox-launcher": "^0.1.6"` in my `package.json`, and followed your lead to call Protractor with `--browser=firefox`. This works!

Now I need a way of accessing my Github Secret on Travis as I'm getting `ReferenceError: gitAccessToken is not defined` because of course my `secrets.js` file isn't pushed up to Travis. I wonder how this is managed "in the wild"? Probably a question for another ticket! :)

Thanks again for your hep,

Tim
